File: tests/helpers/urls.ts
// SPDX-License-Identifier: Apache-2.0

/**
 * Centralized service URLs for tests.
 *
 * Every URL reads from `process.env` first so that the same test suite can be
 * pointed at a remote deployment:
 *
 *   VERIFIER_URL=https://verifier.example.com pnpm run test
 */

export const VERIFIER_URL = process.env.VERIFIER_URL || 'http://localhost:3000';
export const MANAGEMENT_URL =
  process.env.MANAGEMENT_URL || 'http://localhost:3001/v1';
export const MANAGEMENT_ROOT =
  process.env.MANAGEMENT_ROOT || 'http://localhost:3001';
export const AGENT_A_URL = process.env.AGENT_A_URL || 'http://localhost:8787';
export const AGENT_B_URL = process.env.AGENT_B_URL || 'http://localhost:8788';
export const AGENT_C_URL = process.env.AGENT_C_URL || 'http://localhost:8789';

/**
 * Force the Verifier management reporter to flush its buffer immediately,
 * then wait briefly for the management server to persist the entries.
 * Falls back to a fixed 8s wait if the flush endpoint isn't available.
 */
export async function flushVerifierReporter(
  verifierUrl: string,
): Promise<void> {
  try {
    const res = await fetch(`${verifierUrl}/internal/reporter/flush`, {
      method: 'POST',
      signal: AbortSignal.timeout(5000),
    });
    if (res.ok) {
      // Brief pause for management to persist the flushed entries
      await new Promise((r) => setTimeout(r, 2_000));
      return;
    }
  } catch {
    // endpoint not available — fall through
  }
  // Fallback: wait for the periodic 5s flush + buffer
  await new Promise((r) => setTimeout(r, 8_000));
}

/**
 * Returns `true` when the service at `url` responds with HTTP 2xx on `path`.
 */
export async function checkServerRunning(
  url: string,
  path = '/health',
): Promise<boolean> {
  try {
    const response = await fetch(`${url}${path}`, {
      signal: AbortSignal.timeout(5000),
    });
    return response.ok;
  } catch {
    return false;
  }
}

File: tests/helpers_py/urls.py
# SPDX-License-Identifier: Apache-2.0

"""Centralized service URLs and utilities for Python integration tests.

Mirrors tests/helpers/urls.ts.
"""

from __future__ import annotations

import asyncio
import os

import httpx

VERIFIER_URL = os.environ.get("VERIFIER_URL", "http://localhost:3000")
MANAGEMENT_URL = os.environ.get("MANAGEMENT_URL", "http://localhost:3001/v1")
MANAGEMENT_ROOT = os.environ.get("MANAGEMENT_ROOT", "http://localhost:3001")
AGENT_A_URL = os.environ.get("AGENT_A_URL", "http://localhost:8787")
AGENT_B_URL = os.environ.get("AGENT_B_URL", "http://localhost:8788")
AGENT_C_URL = os.environ.get("AGENT_C_URL", "http://localhost:8789")
AGENT_PA_URL = os.environ.get("AGENT_PA_URL", "http://localhost:8801")
AGENT_PB_URL = os.environ.get("AGENT_PB_URL", "http://localhost:8802")
AGENT_PC_URL = os.environ.get("AGENT_PC_URL", "http://localhost:8803")
AGENT_PD_URL = os.environ.get("AGENT_PD_URL", "http://localhost:8804")


async def check_server_running(url: str, path: str = "/health") -> bool:
    """Return True when the service at *url* responds with HTTP 2xx on *path*."""
    try:
        async with httpx.AsyncClient(timeout=5.0) as client:
            resp = await client.get(f"{url}{path}")
            return resp.is_success
    except Exception:
        return False


async def flush_verifier_reporter(verifier_url: str) -> None:
    """Force the Verifier management reporter to flush its buffer immediately.

    Falls back to a fixed 8 s wait if the flush endpoint isn't available.
    """
    try:
        async with httpx.AsyncClient(timeout=5.0) as client:
            resp = await client.post(f"{verifier_url}/internal/reporter/flush")
            if resp.is_success:
                await asyncio.sleep(2)
                return
    except Exception:
        pass
    # Fallback: wait for the periodic 5 s flush + buffer
    await asyncio.sleep(8)


async def chat(agent_url: str, message: str, timeout: float = 120.0) -> str:
    """POST to an agent's /chat endpoint and return the response text."""
    async with httpx.AsyncClient(timeout=timeout) as client:
        resp = await client.post(
            f"{agent_url}/chat",
            json={"message": message},
        )
        resp.raise_for_status()
        data = resp.json()
        return data.get("response") or data.get("message") or str(data)
